src/components/transactions/SingleTx/index.tsx

```diff
@@ -40,21 +40,17 @@ const SingleTx = () => {
   const transactionId = Array.isArray(id) ? id[0] : id
   const { safe, safeAddress } = useSafeInfo()
 
-  const [txDetails, txDetailsError] = useAsync<TransactionDetails>(
-    () => {
-      if (!transactionId || !safeAddress) return
+  const [txDetails, txDetailsError] = useAsync<TransactionDetails>(() => {
+    if (!transactionId || !safeAddress) return
 
-      return getTransactionDetails(safe.chainId, transactionId).then((details) => {
-        // If the transaction is not related to the current safe, throw an error
-        if (!sameAddress(details.safeAddress, safeAddress)) {
-          return Promise.reject(new Error('Transaction with this id was not found in this Safe'))
-        }
-        return details
-      })
-    },
-    [transactionId, safe.chainId, safe.txQueuedTag, safe.txHistoryTag, safeAddress],
-    false,
```

Comment:
Instead of clearing the data (causing a rerender if the tags change), I think we should create a Safe-specific flag along the lines of:

```ts
  const { prefix } = parsePrefixedAddress(router.query.safe)
  const shouldClearData = safe.address.value !== txDetails?.safeAddress || prefix !== chains[safe.chainId]
```

Reply:
After testing this again it seems to clear the transaction fine when switching safes so I reverted it.